executor/batch_point_get.go

```diff
@@ -475,7 +475,8 @@ func (e *BatchPointGetExec) initialize(ctx context.Context) error {
 	for i, key := range keys {
 		val := values[string(key)]
 		if len(val) == 0 {
-			if e.idxInfo != nil && (!e.tblInfo.IsCommonHandle || !e.idxInfo.Primary) {
+			if e.idxInfo != nil && (!e.tblInfo.IsCommonHandle || !e.idxInfo.Primary) &&
```

Comment:
If there is an other reason for data inconsistency, we also ignore it ?

Reply:
`e.ctx.GetSessionVars().StmtCtx.WeakConsistency == true` means it's in a read-only autocommitted query. For other situations, `e.ctx.GetSessionVars().StmtCtx.WeakConsistency` should be `false`.